yarpcerrors/errors.go

```diff
@@ -65,9 +65,12 @@ func FromError(err error) *Status {
 		return st
 	}
 
+	// Extra wrapping ensures Unwrap works consistently across *Status created
+	// by FromError and Newf.
+	// https://github.com/yarpc/yarpc-go/pull/1966
 	return &Status{
 		code: CodeUnknown,
-		err:  err,
+		err:  fmt.Errorf("%w", err),
```

Comment:
This is technically correct but not the most efficient. `fmt.Errorf` will format the string and then wrap the error. We could just wrap the error in the same way fmt does.

See: https://golang.org/src/fmt/errors.go?s=624:674#L7

Reply:
sgtm, @jkanywhere would you mind adding a error wrapper type that does the same thing?